src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/RuntimeConfigurationRootProvider.cs

```diff
@@ -96,6 +96,9 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 // * N times pointer to value
                 builder.EmitNaturalInt(settings.Count);
 
+                // osx-arm64 requires alignment before emitting the following pointers
+                builder.RequireInitialPointerAlignment();
```

Comment:
This comment and placement make it sound like that this is going to insert padding between Count and the Key-Value pairs. That's not correct - the alignment is going to be applied to the whole node.

I would move this call up before `builder.AddSymbol(this);`,, similar to where it is for most other builders.

I would omit mention of osx-arm64 from the comment. This alignment is general goodness, it is a perf optimization at least.